### PR TITLE
[CP-2466] API response validators

### PR DESCRIPTION
**Pull request by lkowalczyk87**
JIRA Reference: [CP-2466]

### :memo: Description ️

-

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2466]: https://appnroll.atlassian.net/browse/CP-2466?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mkurczewski — DISMISSED** [review]
LGTM ✅

File: libs/device/feature/src/lib/api-features/api-features.service.ts [hunk 1]
```diff
@@ -64,7 +67,11 @@ export class APIFeaturesService {
     })
 
     if (response.ok) {
-      return Result.success(response.data.body as OverviewConfig)
+      const config = OverviewConfigValidator.safeParse(response.data.body)
+
+      return config.success
+        ? Result.success(config.data)
+        : Result.failed(new AppError(GeneralError.IncorrectResponse, ""))
```
Comment (dkarski) [inline libs/device/feature/src/lib/api-features/api-features.service.ts:74]: 🟢 
```suggestion
        : Result.failed(new AppError(GeneralError.IncorrectResponse))
```

File: libs/device/feature/src/lib/menu/menu.service.ts [hunk 2]
```diff
@@ -34,7 +38,11 @@ export class APIMenuService {
       },
     })
     if (response.ok) {
-      return Result.success(response.data.body as MenuConfig)
+      const menuConfig = MenuConfigValidator.safeParse(response.data.body)
+
+      return menuConfig.success
+        ? Result.success(menuConfig.data)
+        : Result.failed(new AppError(GeneralError.IncorrectResponse, ""))
```
Comment (dkarski) [inline libs/device/feature/src/lib/menu/menu.service.ts:45]: 🟢
```suggestion
        : Result.failed(new AppError(GeneralError.IncorrectResponse))
```

File: libs/device/feature/src/lib/outbox/outbox.service.ts [hunk 3]
```diff
@@ -28,7 +28,11 @@ export class APIOutboxService {
       method: "GET",
     })
     if (response.ok) {
-      return Result.success(response.data.body as Outbox)
+      const outbox = OutboxValidator.safeParse(response.data.body)
+
+      return outbox.success
+        ? Result.success(outbox.data)
+        : Result.failed(new AppError(GeneralError.IncorrectResponse, ""))
```
Comment (dkarski) [inline libs/device/feature/src/lib/outbox/outbox.service.ts:35]: 🟢
```suggestion
        : Result.failed(new AppError(GeneralError.IncorrectResponse))
```

File: libs/device/models/src/lib/api-config/api-config.test.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  it("should return fail when features array is empty", () => {
+    const apiConfig = { ...minimumApiConfig, features: [] }
+
+    const result = ApiConfigValidator.safeParse(apiConfig)
+    expect(result.success).toBeFalsy()
+  })
+  it.each(["apiVersion", "productId", "vendorId"])(
+    "should return fail when %s is missing",
+    (fieldName) => {
+      const apiConfig: any = { ...minimumApiConfig }
```
Comment (dkarski) [inline libs/device/models/src/lib/api-config/api-config.test.ts:41]: 🟡 
```suggestion
      const apiConfig: ApiConfig = { ...minimumApiConfig }
```

Comment (dkarski) [inline libs/device/models/src/lib/api-config/api-config.test.ts:41]: 🟢 
```suggestion
      const apiConfig: Omit< ApiConfig, "fieldName"> = { ...minimumApiConfig }
```

File: libs/device/models/src/lib/api-config/api-config.test.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      delete apiConfig[fieldName]
```
Comment (dkarski) [inline libs/device/models/src/lib/api-config/api-config.test.ts:42]: ⚪ 
```suggestion
      const { [fieldName]: _, ...apiConfig } = minimumApiConfig;
```

The syntax [fieldName]: _ in object destructuring in JavaScript or TypeScript signifies omitting the value of the key stored in the variable fieldName and assigning that value to the variable _, which is used here as a dummy variable to disregard that particular item while retaining the rest of the object in apiConfig.

TypeScript properly handles this destructuring mechanism, allowing for flexible manipulation of object structures without losing type information for the remaining properties.

**Review by dkarski — DISMISSED** [review]
✅ 

Good job on the work done. I would consider evaluating the trade-offs between using destructuring versus the delete operator for handling types in TypeScript 😉

**Review by mkurczewski — APPROVED** [review]

**Pull request merged**